tests/test_core.py

```diff
@@ -1278,33 +1277,54 @@ def test_exceptions_allowed(self):
     print('empty_size: %d' % empty_size)
     print('fake_size: %d' % fake_size)
     print('disabled_size: %d' % disabled_size)
-    self.assertEqual(empty_size, fake_size)
+    # empty list acts the same a full disable
+    self.assertEqual(empty_size, disabled_size)
     # big change when we disable exception catching of the function
     self.assertGreater(size - empty_size, 0.01 * size)
     # full disable can remove a little bit more
-    self.assertLess(disabled_size, empty_size)
+    self.assertLess(disabled_size, fake_size)
 
   def test_exceptions_allowed_2(self):
-    self.set_setting('DISABLE_EXCEPTION_CATCHING', 2)
     # Wasm does not add an underscore to function names. For wasm, the
     # mismatches are fixed in fixImports() function in JS glue code.
-    self.set_setting('EXCEPTION_CATCHING_ALLOWED', ["main"])
+    self.set_setting('EXCEPTION_CATCHING_ALLOWED', ['main'])
     # otherwise it is inlined and not identified
     self.set_setting('INLINING_LIMIT')
 
     self.do_core_test('test_exceptions_allowed_2.cpp')
 
   def test_exceptions_allowed_uncaught(self):
     self.emcc_args += ['-std=c++11']
-    self.set_setting('DISABLE_EXCEPTION_CATCHING', 2)
     # Wasm does not add an underscore to function names. For wasm, the
     # mismatches are fixed in fixImports() function in JS glue code.
-    self.set_setting('EXCEPTION_CATCHING_ALLOWED', ["_Z4testv"])
+    self.set_setting('EXCEPTION_CATCHING_ALLOWED', ['_Z4testv'])
     # otherwise it is inlined and not identified
     self.set_setting('INLINING_LIMIT')
 
     self.do_core_test('test_exceptions_allowed_uncaught.cpp')
 
+  def test_exceptions_allowed_misuse(self):
+    self.set_setting('EXCEPTION_CATCHING_ALLOWED', ['foo'])
+
+    # Test old =2 setting for DISABLE_EXCEPTION_CATCHING
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 2)
+    err = self.expect_fail([EMCC, test_file('hello_world.c')] + self.get_emcc_args())
+    self.assertContained('error: DISABLE_EXCEPTION_CATCHING is no longer needed when specifying EXCEPTION_CATCHING_ALLOWED [-Wdeprecated] [-Werror]', err)
+
+    # =0 and =2 should be a hard error
```

Comment:
```suggestion
    # =0 and =1 should be a hard error
```
I think having 0 as a warning is more intuitive though.

Reply:
Why?   The definition of =0 today (before this change) means "all functions" .. so specifying additionally an allow list makes not sense today.. so there should be no such users.

Reply:
As of today the only value that makes sense along with the `allow list` is `=2`, right?  Why should I start allowing `=0` + `allow list` in this change?   The point of this change is to try to make these two settings as mutually exclusive as possible without breaking backwards compat.

Reply:
OK, I made =0 into a warning too.  The difference I don't think is very important.

Reply:
I think my interpretation of the switches were different. I thought `DISABLE_EXCEPTION_CATCHING` as a on-off switch, which we turn on whenever (in all functions or in only some of them) we want to catch exceptions, and `EXCEPTION_CATCHING_ALLOWED` as a fine-grained knob to say which function (if not specified, all functions).

I thought this interpretation was more intuitive, but I don't have strong opinions, so please revert if you feel like it.